src/js/components/ServiceList.js

```diff
@@ -35,8 +36,19 @@ let ServiceList = React.createClass({
     return !_.isEqual(this.props, nextProps) || changedState;
   },
 
-  handleServiceClick: function (serviceName) {
-    this.context.router.transitionTo('dashboard-panel', {serviceName});
+  handleServiceClick: function (service, event) {
+    // Open service in new window/tab if service has a web URL
+    if (service.webui_url &&
+      (event.ctrlKey || event.shiftKey || event.metaKey)) {
```

Comment:
This handles if you open a new tab or a new window with a modifier key, i.e. shift, meta (command on mac) or ctrl (for Windows, will still be work as secondary click for Mac). Here is a reference: http://stackoverflow.com/a/20087506/1008519